### ISSUE TITLE
Чек коррекции пример

### ISSUE THREAD

**Issue by susloparov77** [shtrih-m/fr_drv_ng#9]
Здравствуйте не могли бы вы предоставить пример по печати чека коррекции, не смог ни где найти?

**Comment by swex** [comment]
Необходимо воспользоваться методами:
`FNBeginCorrectionReceipt`, `FNBuildCorrectionReceipt2`
В classic_interface.h расписаны используемые свойства. Подход везде общий:
1. Заполняем **ВСЕ** свойства, которые используются методом(именно все, т.к. свойства могут быть изменены выполнением предыдущих команд 
2. Вызываем метод
3. Проверяем код возврата от метода
4. Пользуемся модифицированными свойствами

**Comment by susloparov77** [comment]
А что значит "Работает только с онлайн-кассами"?

**Comment by susloparov77** [comment]
Я извиняюсь но без примера это написать сложно. Вы то этим занимаетесь долго а я неделю)

1. Заполняем ВСЕ свойства, которые используются методом(именно все, т.к. свойства могут быть изменены выполнением предыдущих команд - какие свойства чего записываем куда?

**Comment by Mzales** [comment]
Вот описание методов:  
[FNBeginCorrectionReceipt](../wiki/Methods#group___base_methods_1ga328d7f37ec5e9f659a406b7722089917)
[FNBuildCorrectionReceipt2](../wiki/Methods#group___base_methods_1gaee9e3360c562ab9b43163bc42e9a61bf).   
"Используемые свойства" заполняете до вызова функции, "модифицируемые свойства" используете после вызова.

**Comment by susloparov77** [comment]
Можете прислать любой пример или подскажите место в документации где есть этот пример?

**Comment by Mzales** [comment]
Вот работающий пример:
```c++
    uint32_t adminPassword = 30;
    //начало чека коррекции
    ci.Set_Password(adminPassword);
    ret = ci.FNBeginCorrectionReceipt();
    //отправка тэгов

    //устанавливаем тег 1177
    ci.Set_TagNumber(1177);
    ci.Set_TagType(7);
    ci.Set_TagValueStr("correction name");
    ret = ci.FNSendTag();

    //здесь устанавливаем тег 1178

    //устанавливаем тег 1179
    ci.Set_TagNumber(1179);
    ci.Set_TagType(7);
    ci.Set_TagValueStr("12345");
    ret = ci.FNSendTag();

    //печать чека коррекции
    ci.Set_Password(adminPassword);
    ci.Set_Summ1(3000);//сумма коррекции
    ci.Set_Summ2(0);
    ci.Set_Summ3(0);
    ci.Set_Summ4(0);
    ci.Set_Summ5(0);
    ci.Set_Summ6(0);
    ci.Set_Summ7(0);
    ci.Set_Summ8(0);
    ci.Set_Summ9(0);
    ci.Set_Summ10(0);
    ci.Set_Summ11(0);
    ci.Set_Summ12(0);
    ci.Set_CorrectionType(0);//Тип коррекции "самостоятельно"
    ci.Set_CalculationSign(1);//признак расчета "коррекция прихода"
    ci.Set_TaxType(1);//схема налогообложения "основная"
    ret = ci.FNBuildCorrectionReceipt2();
```

**Comment by susloparov77** [comment]
Спасибо большое))

**Comment by susloparov77** [comment]
Скажите а зачем вы ci.Set_Password(adminPassword); два раза делаете???

**Comment by Mzales** [comment]
Устанавливать два раза одинаковый пароль, конечно, не нужно. Просто хотелось показать, что перед каждой функцией надо не забывать заполнять все используемые ей свойства. Правда, тогда надо было еще и перед [FNSendTag()](../wiki/Methods#group___base_methods_1ga9e5c46c7c8a5d615bc18b2999180c050) заполнить:)

**Comment by susloparov77** [comment]
Ок огромное спасибо, еще бы по ФН закрытие открытие примерчик дали??) Вам надо книгу издать с примерами без них не напишешь, тут надо ядро знать).

**Comment by susloparov77** [comment]
взял пример полностью  здесь execute(std::bind(&classic_interface::FNBuildCorrectionReceipt2, &m_ci));
Ошибка 51 Некорректные параметры команды, что может быть??

**Comment by olefard** [comment]
Выложите, пожалуйста, лог.
Если у вас смарт-пос-ф, то после получения ошибки 51 выполните:
```
m_ci.ReadLastErrorDescription();
auto desc = m_ci.Get_ErrorDescription();// здесь будет подробное описание ошибки, какой именно параметр некорректен.
```

**Comment by susloparov77** [comment]
Что нибудь можете сказать мысли есть??

**Comment by Mzales** [comment]
Да, Summ1 должна равняться сумме всех типов оплаты. К сожалению, не могу проверить на КЯ, на моем ФР все итак работало, хотя и неправильно. Пример исправлен.

**Comment by susloparov77** [comment]
А где можно прочитать что такое Set_Summ1, Set_Summ2 .... Set_SummN  Они чем отличаются??? В 2 всегда наличка?? Потом 3 Электронные деньги??

**Comment by Mzales** [comment]
По-прежнему здесь [FNBuildCorrectionReceipt2](../wiki/Methods#group___base_methods_1gaee9e3360c562ab9b43163bc42e9a61bf).

**Comment by susloparov77** [comment]
А для FNCloseCheckEx Set_Summ1, Set_Summ2 .... Set_SummN Содержат другой смысл по документации, так?

**Comment by Mzales** [comment]
Да, другой. Там Summ1...Summ16 обозначают 16 видов оплаты.  
[FNCloseCheckEx](../wiki/Methods#group___base_methods_1ga8f0dab763204209edd8f164f9ef5493b).

**Comment by susloparov77** [comment]
Спасибо все ок можно закрывать тему.
